backend/src/main/java/com/google/minicrm/data/Form.java

```diff
@@ -92,4 +94,18 @@ public Entity asEntity() {
   public static Key generateKey(Key parentKey, long formId) {
     return KeyFactory.createKey(parentKey, KIND_NAME, formId);
   }
+
+  /**
+   * Same as {@link com.google.minicrm.data.Form#generateKey(Key, long)} but for multiple formIds.
+   * @param user    the user owning the forms
+   * @param formIds the ids of the forms
+   * @return        the keys for the form entitied with the specified formIds and user.
+   */
+  public static Key[] generateKeys(Key parentKey, long[] formIds) {
+    List<Key> keys = new ArrayList<>();
+    for(long formId : formIds) {
+      keys.add(generateKey(parentKey, formId));
+    }
+    return keys.toArray(new Key[0]);
+  }
 }
```

Comment:
Could we just initialize the Key[] array with the size of the formId arrays?

Reply:
No need. The size here doesn't matter. I'm passing an array reference here so that Java compiler can correctly infer the type. I could use `toArray()` (no args), but that returns an Object[] when I need a Key[]. Instead, I chose to use `toArray(T[] arg)` since that will return a `T[]`. You can read up some more about it [here](https://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html#toArray(T[])).

Reply:
does this need to be a primitive array? It looks like it can accept any iterable as well: https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/DatastoreService.html#delete-java.lang.Iterable-. I think it would be preferable to choose one to use as opposed to converting between the two if there's not a clear benefit to it.

Reply:
Good catch. Should be fixed now.